subxt/src/backend/legacy/rpc_methods.rs

```diff
@@ -36,9 +40,13 @@ impl<T: Config> LegacyRpcMethods<T> {
         key: &[u8],
         hash: Option<T::Hash>,
     ) -> Result<Option<StorageData>, Error> {
-        let params = rpc_params![to_hex(key), hash];
-        let data: Option<Bytes> = self.client.request("state_getStorage", params).await?;
-        Ok(data.map(|b| b.0))
+        self.retry
```

Comment:
I'm not sure how I feel about having the retry logic in the `rpc_methods` files. I sortof think that it would be best to keep all of the retry stuff in the Backend impls so that it's really obvious what will be retried and how, and keeping the raw rpc method impls super simple so that they are just the lowest level way to make a basic api call to a node.

Reply:
so I'm quite scared the folks will use legacy stuff through the rpc methods and then miss the retry logic but I agree it would be nicer to have it just in the backend impls

Reply:
if you have a look at the legacy stuff then lots of stuff will not have retry logic if we don't add to the rpc methods

Reply:
I have removed the retry logic for the rpc methods stuff for now and the idea is to expose a helper function `retry_with_policy` that should be easy to use...